netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -570,17 +570,22 @@ private void createStream(CreateStreamCommand command, ChannelPromise promise)
       return;
     }
     if (connection().goAwayReceived()) {
+      Status s = abruptGoAwayStatus;
+      int maxActiveStreams = connection().local().maxActiveStreams();
+      if (s == null) {
+        // Should be impossible, but handle pseudo-gracefully
+        s = Status.INTERNAL.withDescription(
+            "Failed due to abrupt GOAWAY, but can't find GOAWAY details");
+      } else if (streamId > connection().local().lastStreamKnownByPeer()) {
+        s = s.augmentDescription("stream id: " + streamId);
```

Comment:
The stream id isn't directly useful without something to compare it to. Either just don't have this condition or also include lastStreamKnownByPeer in the message.

Reply:
I avoided adding `lastStreamKnownByPeer` because I thought it is included in GOAWAY debug data.

Reply:
It is not part of "Additional Debug Data" of [the GOAWAY frame](https://httpwg.org/specs/rfc7540.html#GOAWAY). "debug data" in the Netty API revers to that data. We normally don't include it because it shouldn't really matter. But it could be okay to include here.

Reply:
Seems not, lastStreamId, errorCode, and debugData are separated.

Reply:
Added .